src/components/molecules/input/Input.tsx

```diff
@@ -0,0 +1,97 @@
+import { Input as BaseInput } from "@base-ui-components/react";
+import { Icon, IconName, iconNames } from "@components/atoms/icon/Icon";
+import { joinTailwindClasses } from "@utils/tailwind";
+import { Dispatch, SetStateAction, useMemo } from "react";
+
+interface InputProps extends Omit<React.InputHTMLAttributes<HTMLInputElement>, "size"> {
+  clearable?: boolean;
+  containerClasses?: string;
+  color?: "brand" | "mono";
+  icon?: IconName | React.ReactNode;
+  iconOnLeft?: boolean;
+  inputClasses?: string;
+  size?: "small" | "medium" | "large";
+  valueSetter?: Dispatch<SetStateAction<string>>;
```

Comment:
Question: is this used only for the clear event?

Reply:
Yes - I played with a few other ways of dealing with the clear event, but settled on this. If you want the clear button to work, you need to pass your `useState` setter function into this prop.

Having a less opinionated input that you can totally customise or use for real edge cases was preferable to a locked-down component with its own internal state.